src/app/contexts/RequestContext/index.jsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { node, string, func } from 'prop-types';
+import getOriginContext from './getOriginContext';
+
+const RequestContext = React.createContext('default');
+
+export const RequestContextProvider = ({ children, platform, bbcOrigin }) => {
+  const value = {
+    platform,
+    ...getOriginContext(bbcOrigin),
+  };
+
+  return (
+    <RequestContext.Provider value={value}>{children}</RequestContext.Provider>
+  );
+};
+
+export const RequestContextConsumer = ({ children }) => (
+  <RequestContext.Consumer>{children}</RequestContext.Consumer>
+);
+
+RequestContextProvider.propTypes = {
+  children: node.isRequired,
+  platform: string.isRequired,
+  bbcOrigin: string,
+};
+
+RequestContextProvider.defaultProps = {
+  bbcOrigin: null,
```

Comment:
What's the reason for not defaulting the `bbcOrigin` here to `https://www.bbc.co.uk`, so it is defined all the way down in the application?

Reply:
I don't want to default, as `null` is a valid input (on the client) and I want the defaulting logic to happen in getOriginContext https://github.com/bbc/simorgh/pull/1477/files/9ddb72139cac0fd9388ab80ae740e5d756b60cf2#diff-3578d7c81c4a8fad428c02d2a9377ff6R2
If it defaults here, it will always use that over the window location on the client

Reply:
Okay, sounds good. Thanks!